test/validation/api/pool/pool.c

```diff
@@ -1121,6 +1130,356 @@ static void pool_test_timeout_pool_statistics(void)
+			       void *buf[], uint32_t num, uint32_t buf_size)
+{
+	odp_shm_t shm;
+	uint8_t *buf_ptr;
+	uint32_t i;
+	uint32_t shm_size, mem_align;
+	uint32_t flags = 0;
+	uint32_t buf_align = EXT_BUF_ALIGN;
+	uint32_t min_align = capa->pkt.min_buf_align;
+
+	CU_ASSERT_FATAL(min_align > 0);
+
+	if (min_align > buf_align)
+		buf_align = min_align;
+
+	if (capa->pkt.buf_size_aligned) {
+		buf_align = buf_size;
+		CU_ASSERT_FATAL((buf_size % min_align) == 0);
+	}
+
+	mem_align = buf_align;
+	if (capa->pkt.min_mem_align > mem_align)
+		mem_align = capa->pkt.min_mem_align;
+
+	/* Prepare to align every buffer */
+	shm_size = (num + 1) * (buf_size + buf_align);
+
+	shm = odp_shm_reserve("test_pool_ext_populate", shm_size, mem_align, 0);
+	if (shm == ODP_SHM_INVALID)
+		return ODP_SHM_INVALID;
+
+	buf_ptr = odp_shm_addr(shm);
+	CU_ASSERT_FATAL((uintptr_t)buf_ptr % mem_align == 0);
+
+	/* Move from mem_align to buf_align */
+	while ((uintptr_t)buf_ptr % buf_align)
+		buf_ptr++;
+
+	for (i = 0; i < num; i++) {
+		if (i == num - 1)
+			flags = ODP_POOL_POPULATE_DONE;
+
+		buf[i] = buf_ptr;
+		CU_ASSERT_FATAL(odp_pool_ext_populate(pool, &buf[i], buf_size, 1, flags) == 0);
+
+		buf_ptr += buf_size;
+		while ((uintptr_t)buf_ptr % buf_align)
+			buf_ptr++;
+	}
+
+	return shm;
+}
+
+static void test_packet_pool_ext_populate(void)
+{
+	odp_shm_t shm;
+	odp_pool_t pool;
+	odp_pool_ext_param_t param;
+	odp_pool_ext_capability_t capa;
+	uint32_t buf_size, num_buf;
+	void *buf[EXT_NUM_BUF];
+
+	CU_ASSERT_FATAL(odp_pool_ext_capability(ODP_POOL_PACKET, &capa) == 0);
+
+	pool_ext_init_packet_pool_param(&param);
+	num_buf  = param.pkt.num_buf;
+	buf_size = param.pkt.buf_size;
+
+	CU_ASSERT_FATAL(capa.pkt.min_head_align > 0);
+
+	pool = odp_pool_ext_create("pool_ext_0", &param);
+	CU_ASSERT_FATAL(pool != ODP_POOL_INVALID);
+
+	shm = populate_pool(pool, &capa, buf, num_buf, buf_size);
+	CU_ASSERT_FATAL(shm != ODP_SHM_INVALID);
+
+	CU_ASSERT(odp_pool_destroy(pool) == 0);
+	CU_ASSERT(odp_shm_free(shm) == 0);
+}
+
+static uint32_t find_buf(odp_packet_t pkt, void *buf[], uint32_t num, uint32_t head_offset)
+{
+	uint32_t i;
+	uint8_t *ptr;
+	uint8_t *head = odp_packet_head(pkt);
+
+	for (i = 0; i < num; i++) {
+		ptr  = buf[i];
+		ptr += head_offset;
+
+		if (head == ptr)
+			break;
+	}
+
+	return i;
+}
+
+#define PKT_LEN_NORMAL    0
+#define PKT_LEN_MAX       1
+#define PKT_LEN_SEGMENTED 2
+
+static void packet_pool_ext_alloc(int len_test)
+{
+	odp_shm_t shm;
+	odp_pool_t pool;
+	odp_pool_ext_param_t param;
+	odp_pool_ext_capability_t capa;
+	uint32_t i, buf_size, num_buf, num_pkt, num_alloc, buf_index;
+	uint32_t pkt_len, head_offset, trailer_size, headroom, max_headroom;
+	uint32_t hr, tr, uarea_size, max_payload, buf_data_size;
+	int num_seg;
+	void *buf[EXT_NUM_BUF];
+	odp_packet_t pkt[EXT_NUM_BUF];
+	uint32_t seg_len = 0;
+
+	CU_ASSERT_FATAL(odp_pool_ext_capability(ODP_POOL_PACKET, &capa) == 0);
+
+	if (capa.pkt.max_segs_per_pkt < 2)
+		return;
```

Comment:
It would be easier to notice from the test output log that segmentation is not supported if `test_packet_pool_ext_alloc_seg()` test would be conditional. Also, now it seems `PKT_LEN_NORMAL` and `PKT_LEN_MAX` tests are also skipped. Is this correct?

Reply:
Fixed in v15. Segmentation testing is conditional.